### ISSUE TITLE
CreateNodeIterator may throw InvalidOperationException

### ISSUE THREAD

**Issue by meziantou** [AngleSharp/AngleSharp#509]
Calling `CreateNodeIterator` throws an unexpected exception when no node match the filter criteria.

```
var config = Configuration.Default.WithDefaultLoader();
var browsingContext = BrowsingContext.New(config);
var content = "<div></div>";
var htmlDocument = await browsingContext.OpenAsync(action => action.Content(content));

var iterator = htmlDocument.CreateNodeIterator(htmlDocument, FilterSettings.Comment); // throw InvalidOperationException
```

**Comment by FlorianRappl** [comment]
Thanks for catching this. Will be fixed in the next release!

While I think the implementation should handle this case (a document instead of an element presented) I also think that you misused the API (you potentially wanted to use `htmlDocument.DocumentElement` or `htmlDocument.Body` as the root); maybe intentionally.

**Comment by meziantou** [comment]
Thanks for the clarification. In fact, this code is just the minimal code to reproduce the exception. In my actual code I use a specific node of the DOM.

**Comment by FlorianRappl** [comment]
I see - yep. The problem (bug) is that the exception is thrown when no child nodes exist. Shouldn't be and will be fixed.

**Comment by FlorianRappl** [comment]
Fixed, see PR #511.
